tox.ini

```diff
@@ -1,39 +1,38 @@
 [tox]
 envlist =
-    py27,py35,py36,py37,pypy,pypy3,coverage,docs
+    py27,py35,py36,py37,py38,pypy,pypy3,coverage,docs
 # Note: if you add new Python versions, please add them to
 # [testenv:coverage] depends as well!
 
 [testenv]
 usedevelop = true
 commands =
-    coverage run -m zope.testrunner --test-path=src []
+    coverage run -p -m zope.testrunner --test-path=src []
+extras =
+    test
 deps =
-    .[test]
     coverage
-setenv =
-   COVERAGE_FILE=.coverage.{envname}
+# Not yet, pending https://github.com/zopefoundation/zope.publisher/issues/49
+# setenv =
+#     ZOPE_INTERFACE_STRICT_IRO=1
 
 [testenv:coverage]
-setenv =
-  COVERAGE_FILE=.coverage
 skip_install = true
 commands =
     coverage erase
     coverage combine
-    coverage html
-    coverage xml
+    coverage html -i
+    coverage xml -i
     coverage report --fail-under=100 --show-missing
 # parallel mode: make sure all builds complete before we run this one
 depends =
-    py27,py35,py36,py37,pypy,pypy3
+    py27,py35,py36,py37,py38,pypy,pypy3
 parallel_show_output = true
 
 [testenv:docs]
 basepython =
-    python3.6
+    python3
 commands =
     sphinx-build -b html -d docs/_build/doctrees docs docs/_build/html
-deps =
-    {[testenv]deps}
-    .[docs]
+extras =
+    docs
```

Comment:
This is a slight change: before tox -e docs would install .[test,docs], now it only installs .[docs].

I think it's an improvement, provided it works (and I trust you tested that it works).

Reply:
Good point. I saw that too and agree it's an improvement (since we're not running doctests).

I did indeed test it.

```console
$ tox -p auto
✔ OK docs in 2.488 seconds
✔ OK py37 in 3.119 seconds
✔ OK py38 in 3.141 seconds
✔ OK py36 in 3.166 seconds
✔ OK py35 in 3.257 seconds
✔ OK py27 in 3.445 seconds
✔ OK pypy in 9.369 seconds
✔ OK pypy3 in 12.181 seconds
✔ OK coverage in 1.61 seconds
coverage installed: coverage==5.0.4
coverage run-test-pre: PYTHONHASHSEED='1142588'
coverage run-test: commands[0] | coverage erase
coverage run-test: commands[1] | coverage combine
coverage run-test: commands[2] | coverage html -i
coverage run-test: commands[3] | coverage xml -i
coverage run-test: commands[4] | coverage report --fail-under=100 --show-missing
Name                                                     Stmts   Miss     Cover   Missing
-----------------------------------------------------------------------------------------
src/zope/traversing/__init__.py                              0      0   100.00%
src/zope/traversing/adapters.py                             80      0   100.00%
src/zope/traversing/api.py                                  88      0   100.00%
src/zope/traversing/browser/__init__.py                      4      0   100.00%
src/zope/traversing/browser/absoluteurl.py                  87      0   100.00%
src/zope/traversing/browser/interfaces.py                   10      0   100.00%
src/zope/traversing/browser/tests.py                       208      0   100.00%
src/zope/traversing/interfaces.py                           33      0   100.00%
src/zope/traversing/namespace.py                           167      0   100.00%
src/zope/traversing/publicationtraverse.py                  65      0   100.00%
src/zope/traversing/testing.py                              42      0   100.00%
src/zope/traversing/tests/__init__.py                        0      0   100.00%
src/zope/traversing/tests/test_conveniencefunctions.py     175      0   100.00%
src/zope/traversing/tests/test_dependencies.py              18      0   100.00%
src/zope/traversing/tests/test_lang.py                      31      0   100.00%
src/zope/traversing/tests/test_namespacetrversal.py         86      0   100.00%
src/zope/traversing/tests/test_presentation.py              29      0   100.00%
src/zope/traversing/tests/test_publicationtraverse.py      174      0   100.00%
src/zope/traversing/tests/test_skin.py                      31      0   100.00%
src/zope/traversing/tests/test_traverser.py                199      0   100.00%
src/zope/traversing/tests/test_vh.py                        53      0   100.00%
src/zope/traversing/tests/test_vhosting.py                 202      0   100.00%
-----------------------------------------------------------------------------------------
TOTAL                                                     1782      0   100.00%

```